### ISSUE TITLE
nil pointer exception running on android

### ISSUE THREAD

**Issue by C-ollins** [dgraph-io/badger#716]
Golang version: 1.11
Badger: 2.0-rc1

Creating a transaction results in a crash, same code works fine on iOS
```
func runBadger(data string){
	opts := badger.DefaultOptions
	opts.Dir = filepath.Join(data, "/badger")
	opts.ValueDir = filepath.Join(data, "/badger")
	opts.ValueLogLoadingMode = options.FileIO
	opts.TableLoadingMode = options.MemoryMap
	opts.ValueLogFileSize = 209715200
	opts.MaxTableSize = 40000000
	opts.LevelOneSize = 209715200
	opts.NumMemtables = 1
	opts.NumCompactors = 1
	opts.NumLevelZeroTables = 1
	opts.NumLevelZeroTablesStall = 2

	badgerDb, err := badger.Open(opts)
	if err != nil {
		panic(err)
	}

	badgerDb.NewTransaction(true) // crashes here
	println("Transaction created successfully") // this never shows on android
}

```

Crash log
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0xced6deab]
goroutine 17 [running, locked to thread]:
runtime/internal/atomic.Store64(0x9e8c21e4, 0x0, 0x0)
	/usr/local/go/src/runtime/internal/atomic/asm_386.s:199 +0xb
github.com/raedahgroup/dcrlibwallet/vendor/github.com/dgraph-io/badger/y.(*WaterMark).Begin(0x9e8c21d0, 0x0, 0x0)
	/Users/collins/go/src/github.com/raedahgroup/dcrlibwallet/vendor/github.com/dgraph-io/badger/y/watermark.go:74 +0x30
github.com/raedahgroup/dcrlibwallet/vendor/github.com/dgraph-io/badger.(*oracle).readTs(0x9e8c2180, 0x9e8e0000, 0x5b8d80)
	/Users/collins/go/src/github.com/raedahgroup/dcrlibwallet/vendor/github.com/dgraph-io/badger/transaction.go:102 +0x72
github.com/raedahgroup/dcrlibwallet/vendor/github.com/dgraph-io/badger.(*DB).NewTransaction(0x9ec4a000, 0x1, 0x9e858240)
	/Users/collins/go/src/github.com/raedahgroup/dcrlibwallet/vendor/github.com/dgraph-io/badger/transaction.go:659 +0xad
github.com/raedahgroup/dcrlibwallet.runBadger(0x9e858080, 0x37)
	/Users/collins/go/src/github.com/raedahgroup/dcrlibwallet/dcrlibwallet.go:110 +0x1af
github.com/raedahgroup/dcrlibwallet.NewLibWallet(0x9e858080, 0x37, 0x9e888000, 0x8, 0x9e888008, 0x8, 0x9e828000, 0xced756c9, 0x9e858030
	/Users/collins/go/src/github.com/raedahgroup/dcrlibwallet/dcrlibwallet.go:86 +0x336
main.proxydcrlibwallet__NewLibWallet(0xeaa700c0, 0x37, 0xea998540, 0x8, 0xea998460, 0x8, 0x8, 0xcedbf5d7)
	/var/folders/_z/21zf85t530b1wz5mzpmrqrkc0000gp/T/gomobile-work-433704709/src/gobind/go_dcrlibwalletmain.go:2009 +0xa3
main._cgoexpwrap_dc71ff73588f_proxydcrlibwallet__NewLibWallet(0xeaa700c0, 0x37, 0xea998540, 0x8, 0xea998460, 0x8, 0xeaa74900, 0xea9f400
	_cgo_gotypes.go:3169 +0x43
```

**Comment by C-ollins** [comment]
Any solution to this crash yet?

**Comment by manishrjain** [comment]
@jarifibrahim -- Can you take a look?

**Comment by jarifibrahim** [comment]
Hey @C-ollins :wave:, how can I reproduce this failure?

**Comment by C-ollins** [comment]
Sorry I'm just seeing this, I thought this got abandoned. You can run it by compiling the code I wrote with Gomobile and running on an android. I'm available to help at any part.

**Comment by C-ollins** [comment]
Hello @jarifibrahim, still looking into this?

**Comment by jarifibrahim** [comment]
@C-ollins I haven't had a chance to look at this. I'll look into it today (or tomorrow).

**Comment by jarifibrahim** [comment]
@C-ollins I tried to build the code snippet you've given but can't seem to build it.
```
gopher@1b701f2d047e:~/src/github.com/jarifibrahim/foo$ gomobile build
gomobile: github.com/jarifibrahim/foo does not import "golang.org/x/mobile/app"
```
I think I'll need more than just the code you've shared to build it with `gomobile`.
I've never used `gomobile` before. Can you please give me some more details on how can I reproduce the failure?

**Comment by C-ollins** [comment]
Alright, I'll create a repository with everything you need to run the app. All you'd need is android studio to compile the app.

**Comment by jarifibrahim** [comment]
@C-ollins The issue you're facing has been fixed in v2.0.0-rc.2 via https://github.com/dgraph-io/badger/issues/610

Please try `v2.0.0-rc.2` (or master). it should work.

**Comment by jarifibrahim** [comment]
I'm closing this issue for now. Feel free to reopen.
